rotkehlchen/chain/evm/decoding/odos/v2/decoder.py

```diff
@@ -0,0 +1,228 @@
+        for event in context.decoded_events:
+            if (
+                ((input_amount := input_tokens.get(event.asset.identifier)) == event.balance.amount) or  # noqa: E501
+                input_amount == ZERO  # this means https://docs.odos.xyz/product/sor/v2/#max-balance-swapping  # noqa: E501
+            ) and (
+                (
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ) or (
+                    event.event_type == HistoryEventType.TRADE and
+                    event.event_subtype == HistoryEventSubType.SPEND
+                )
+            ):
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.SPEND
+                event.notes = f'Swap {event.balance.amount} {event.asset.symbol_or_name()} in Odos v1'  # noqa: E501
+                event.event_type = HistoryEventType.TRADE
+                event.address = self.router_address
+                event.counterparty = CPT_ODOS_V2
+                out_events.append(event)
+
+            elif output_tokens.get(event.asset.identifier) == event.balance.amount and (
+                (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ) or (
+                    event.event_type == HistoryEventType.TRADE and
+                    event.event_subtype == HistoryEventSubType.RECEIVE
+                )
+            ):
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.RECEIVE
+                event.notes = f'Receive {event.balance.amount} {event.asset.symbol_or_name()} as the result of a swap in Odos v1'  # noqa: E501
+                event.address = self.router_address
+                event.counterparty = CPT_ODOS_V2
+                in_events.append(event)
+
+        # fees are the amount of out tokens that router received - the amount that router sent
+        router_holding_amount: defaultdict[Asset, FVal] = defaultdict(FVal)
+        for tx_log in context.all_logs:
+            if (
+                tx_log.topics[0] != ERC20_OR_ERC721_TRANSFER or
+                ((identifier := evm_address_to_identifier(
+                    address=tx_log.address,
+                    chain_id=self.evm_inquirer.chain_id,
+                    token_type=EvmTokenKind.ERC20,
+                )) not in output_tokens)
+            ):
+                continue
+
+            # can directly initialize EvmToken here because it's in output_tokens which were get_or_created above  # noqa: E501
+            asset = EvmToken(identifier)
+
+            if hex_or_bytes_to_address(tx_log.topics[2]) == self.router_address:
+                router_holding_amount[asset] += FVal(hex_or_bytes_to_int(tx_log.data)) / FVal(10 ** asset.get_decimals())  # noqa: E501
```

Comment:
since it is an evm token and not a crypto asset you can do always `.decimals`. No need for extra call right?